panel/lxqtpanel.cpp

```diff
@@ -664,62 +764,104 @@ void LXQtPanel::updateWmStrut()
+                                             );
 
-            break;
+                break;
 
-        case LXQtPanel::PositionRight:
+            case LXQtPanel::PositionRight:
+                KX11Extras::setExtendedStrut(wid,
+                                             /* Left   */  0, 0, 0,
+                                             /* Right  */  wholeScreen.right() - rect.right() + getReserveDimension(), rect.top(), rect.bottom(),
+                                             /* Top    */  0, 0, 0,
+                                             /* Bottom */  0, 0, 0
+                                             );
+                break;
+            }
+        } else
+        {
             KX11Extras::setExtendedStrut(wid,
-                                            /* Left   */  0, 0, 0,
-                                            /* Right  */  wholeScreen.right() - rect.right() + getReserveDimension(), rect.top(), rect.bottom(),
-                                            /* Top    */  0, 0, 0,
-                                            /* Bottom */  0, 0, 0
-                                           );
-            break;
+                                         /* Left   */  0, 0, 0,
+                                         /* Right  */  0, 0, 0,
+                                         /* Top    */  0, 0, 0,
+                                         /* Bottom */  0, 0, 0
+                                         );
+        }
     }
-    } else
+    else if(qGuiApp->nativeInterface<QNativeInterface::QWaylandApplication>())
     {
-        KX11Extras::setExtendedStrut(wid,
-                                        /* Left   */  0, 0, 0,
-                                        /* Right  */  0, 0, 0,
-                                        /* Top    */  0, 0, 0,
-                                        /* Bottom */  0, 0, 0
-                                       );
+        //TODO: duplicated code, also set in setPanelGeometry()
+
+        if (mReserveSpace)
+        {
+            LayerShellQt::Window::Anchor edge;
+
+            switch (mPosition)
+            {
+            case LXQtPanel::PositionTop:
+                edge = LayerShellQt::Window::AnchorTop;
+                break;
+
+            case LXQtPanel::PositionBottom:
+                edge = LayerShellQt::Window::AnchorBottom;
+                break;
+
+            case LXQtPanel::PositionLeft:
+                edge = LayerShellQt::Window::AnchorLeft;
+                break;
+
+            case LXQtPanel::PositionRight:
+                edge = LayerShellQt::Window::AnchorRight;
+                break;
+            }
+
+            mLayerWindow->setExclusiveEdge(edge);
+            mLayerWindow->setExclusiveZone(getReserveDimension());
+        }
+        else
+        {
+            mLayerWindow->setExclusiveEdge(LayerShellQt::Window::AnchorNone);
+            mLayerWindow->setExclusiveZone(0);
+        }
```

Comment:
Nullity check for `mLayerWindow`, as is done in other places?

Reply:
Done.